tests/test_native_serve.py

```
@@ -107,6 +159,81 @@ def check(self, name: str, cond: bool, detail: str = ""):
+                 port: int, extra_args: list[str] | None = None,
+                 eager: bool = True, start_timeout: float = 10.0):
+        self.host, self.port = host, port
+        self.log: deque[str] = deque(maxlen=4000)
+        args = [str(binary), "--model", model, "--gguf", gguf,
+                "--host", host, "--port", str(port)]
+        if not eager:
+            args.append("--no-eager-load")
+        args += extra_args or []
+        print(f"Starting {binary.name} on {host}:{port} "
+              f"({'eager' if eager else 'lazy'}) extra={extra_args or []}")
+        self.proc = subprocess.Popen(args, stdin=subprocess.DEVNULL,
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)